### PR TITLE
Update vaughnlive.py

### PR DESCRIPTION
**Pull request by cirrusUK**
here we go again ...

### CODE REVIEW

File: src/streamlink/plugins/vaughnlive.py [hunk 1]
```diff
@@ -1,17 +1,27 @@
 import re
-
+ 
 from streamlink import StreamError
 from streamlink.plugin import Plugin
-from streamlink.plugin.api import useragents
+from streamlink.plugin.api import http, validate, useragents
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: don't use `http` use `self.session.http`

File: src/streamlink/plugins/vaughnlive.py [hunk 2]
```diff
@@ -1,17 +1,27 @@
 import re
-
+
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: remove the spaces

File: src/streamlink/plugins/vaughnlive.py [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
+#   /(?P<channel>[^/&?]+)
+#""", re.VERBOSE)
 
 _url_re = re.compile(r"""
-    http(s)?://(\w+\.)?
-    (?P<domain>vaughnlive|breakers|instagib|vapers|pearltime).tv
-    (/embed/video)?
-    /(?P<channel>[^/&?]+)
+   http(s)?://(\w+\.)?
+   (?P<domain>vaughn).live
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: you need to add some tests

https://github.com/streamlink/streamlink/blob/master/tests/plugins/test_vaughnlive.py

---

update https://github.com/streamlink/streamlink/blob/master/docs/plugin_matrix.rst

---

also the support for all the other domains is missing now

File: src/streamlink/plugins/vaughnlive.py [hunk 4]
```diff
@@ -1,17 +1,27 @@
 import re
-
+ 
 from streamlink import StreamError
 from streamlink.plugin import Plugin
-from streamlink.plugin.api import useragents
+from streamlink.plugin.api import http, validate, useragents
 from streamlink.stream import HTTPStream, StreamIOIterWrapper, StreamIOThreadWrapper
+ 
+#_url_re = re.compile(r"""
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: if not needed, remove it

File: src/streamlink/plugins/vaughnlive.py [hunk 5]
```diff
@@ -1,17 +1,27 @@
...
-    (/embed/video)?
-    /(?P<channel>[^/&?]+)
+   http(s)?://(\w+\.)?
+   (?P<domain>vaughn).live
+   (/embed/video)?
+   /(?P<channel>[^/&?]+)
 """, re.VERBOSE)
 
+_schema = validate.Schema(
+    validate.text
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: just a `validate.text` probably not needed

File: src/streamlink/plugins/vaughnlive.py [hunk 6]
```diff
@@ -22,53 +32,63 @@ def open(self):
...
                                         timeout=timeout,
                                         **self.args)
-
+ 
         def fix_stream():
             """
-            Replace the first 3 bytes of the stream with b'FLV'
-            :return: stream iterator
-            """
+           Replace the first 3 bytes of the stream with b'FLV'
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: readd the spaces here

File: src/streamlink/plugins/vaughnlive.py [hunk 7]
```diff
@@ -22,53 +32,63 @@ def open(self):
...
-            "User-Agent": useragents.FIREFOX
+            "User-Agent": useragents.FIREFOX,
         }
         m = _url_re.match(self.url)
         if m:
             stream_name = "{0}_{1}".format(self.domain_map[(m.group("domain").lower())],
                                            m.group("channel"))
+
+            res = http.get(self.url, schema=_schema)
+            if not res: return
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: probably not needed

File: src/streamlink/plugins/vaughnlive.py [hunk 8]
```diff
@@ -22,53 +32,63 @@ def open(self):
...
+
+            pattern = r"""var mp4StreamUrl = \"(.*?)&secure=(.*?)&expires=(.*?)&ip=(.*?)\""""
+            r = re.search(pattern, res)
+            if not r: return
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: split into two lines and add some `log.error`

File: src/streamlink/plugins/vaughnlive.py [hunk 9]
```diff
@@ -22,53 +32,63 @@ def open(self):
...
         }
         m = _url_re.match(self.url)
         if m:
             stream_name = "{0}_{1}".format(self.domain_map[(m.group("domain").lower())],
                                            m.group("channel"))
+
+            res = http.get(self.url, schema=_schema)
+            if not res: return
+
+            pattern = r"""var mp4StreamUrl = \"(.*?)&secure=(.*?)&expires=(.*?)&ip=(.*?)\""""
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: use `re.compile` and put it somewhere in `class VaughnLive` not here

File: src/streamlink/plugins/vaughnlive.py [hunk 10]
```diff
@@ -22,53 +32,63 @@ def open(self):
...
+            r = re.search(pattern, res)
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: reuse the compiled pattern, not `re.`

File: src/streamlink/plugins/vaughnlive.py [hunk 11]
```diff
@@ -22,53 +32,63 @@ def open(self):
...
+            if not r: return
+
+            stream_link = '%s/play/%s.flv?&secure=%s&expires=%s&ip=%s' % (self.stream_url, stream_name, r.group(2), r.group(3), r.group(4))
```
Comment (back-to) [inline src/streamlink/plugins/vaughnlive.py]: you should use `.format` as this is the most used method in this project

**Comment by cirrusUK** [comment]
Thanks for the help back-to

**Comment by cirrusUK** [comment]
51a6cb9

**Comment by cirrusUK** [comment]
do i now close this PR and start a new pull request with the new .py file with requested changes made ??
sorry for noob noise

**Comment by skulblakka** [comment]
>do i now close this PR and start a new pull request with the new .py file with requested changes made

No need to close and create a new one. Just push the changes to the source branch of the PR and it will automatically update and display them here. (For example if you scroll up a bit it says "cirrusUK added some commits")

**Comment by back-to** [comment]
closing, currently not possible with this solution.

#2178

**Pull request closed without merging**